src/stale-prs.ts

```diff
@@ -319,26 +317,27 @@ export class StalePrFinder {
   }
 
   /**
-   * List the comments and find the last time we marked this PR as stale.
+   * Get the latest warning where we marked this PR as stale.
    */
-  private async mostRecentWarnings(pull_number: number): Promise<Map<Marker, Date>> {
+  private async getLatestWarning(pull_number: number): Promise<Warning | undefined> {
     const comments = await this.client.paginate(this.client.rest.issues.listComments, {
       ...this.repo,
       issue_number: pull_number,
     });
```

Comment:
The API cannot be relied return the comments in order of date. It's not explicitly told to do so in the API or the docs.  The comments can be sorted by date either in the API itself (if possible) or by sorting in the script itself.